HashMapSort/src/main/java/com/AndrieievYurii/UniqueCharCounter.java

```diff
@@ -0,0 +1,43 @@
+package com.AndrieievYurii;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class UniqueCharCounter {
+
+    static Map<String, HashMap<Character,Integer>> cachedWords = new HashMap<>();
+
+    public static HashMap<String, HashMap<Character, Integer>> getCachedWords() {
+        return (HashMap<String, HashMap<Character, Integer>>) cachedWords;
+    }
+
+    @Override
+    public String toString() {
+        return "HashMapCach{" +
+                "cachedWords=" + cachedWords +
+                '}';
+    }
+
+    public Map<Character, Integer> calculateUniqueChars(String input ) {
+        Map<String,HashMap<Character,Integer>> cachedWords = getCachedWords();
+        Map<Character, Integer> charToCount = new HashMap<>();
+        String inputUpperCase = input.toUpperCase();
+        String[] inputSplit = inputUpperCase.split(" ");
+        if (!cachedWords.containsKey(input)) {
+            for(int i = 0; i < inputSplit.length; i++) {
+                char[] inputArray = inputSplit[i].toCharArray();
+                for (char array : inputArray) {
+                    if (!charToCount.containsKey(array)) {
+                        charToCount.put(array, 0);
+                    }
+                    charToCount.put(array, charToCount.get(array) + 1);
+                }
+            }
+            cachedWords.put(input, (HashMap<Character, Integer>) charToCount);
+            return (HashMap<Character, Integer>) charToCount;
+        }else if (cachedWords.containsKey(input)){
```

Comment:
better do vice versa. Better just check if cachedWords.containsKey and another code goes below this if